lib/kafka_rest_client.js

```diff
@@ -100,16 +103,29 @@ KafkaRestClient.prototype.getProduceInterval = function getProduceInterval(retry
 KafkaRestClient.prototype.scheduleTopicMapRefresh = function scheduleTopicMapRefresh() {
     var self = this;
     if (self.proxyRefreshTime > 0) {
-        // console.log('Trying to schedule topic refresh job every: ' +
-        // self.proxyRefreshTime / 1000 + ' seconds.');
         /* eslint-disable no-undef,block-scoped-var */
         self.refreshIntervalId = setInterval(function refreshTopics() {
+            self.scheduleConnect(0);
+        }, self.proxyRefreshTime);
+        /* eslint-enable no-undef,block-scoped-var */
+    }
+};
+
+// Reconnect to rest proxy if scheduleTopicMapRefresh fails
+KafkaRestClient.prototype.scheduleConnect = function scheduleConnect(retry) {
+    var self = this;
+    if (retry < connectWaitTimeMs.length) {
```

Comment:
@xhl1988 I am looking at the code, and It looks like we will only write to the localproxy if we have successfully connected at one point, right?
` if (self.enable) {
        if (produceMessage.topic in self.cachedTopicToUrlMapping) {
            var pathUrl = self.cachedTopicToUrlMapping[produceMessage.topic];
            if (retry === self.maxRetries - 1 && self.maxRetries > 1 && self.isHeatPipeTopic(produceMessage.topic)) {
                pathUrl = self.localAgentHost + ':' + self.localAgentPort;
            }`

That means that the situation where we fail to initially correct would cause failure to produce for at least 30 minutes